### DISCUSSION TITLE
ESP32-S3 pinout: JTAG pins as GPIO

### DISCUSSION THREAD

**Discussion by DustinWatts** [atomic14/esp32-s3-pinouts#8] [General]
Hi Guys and girls, I was hoping someone could tell me how do what I want 😄. Or if it is even possible. 

I understand that if you choose the option to use USB as USB and not JTAG in the Arduino IDE, GPIO39, GPIO40, GPIO41, GPIO42 will automatically become JTAG pins. But what if I want no JTAG at all. I want USB (as a HID) and want to use the GPIO pins as GPIO.

Does anyone have any idea how to do this? Is there API I can use to do this? Or is just `pinMode(39, OUTPUT);` enough? I could find it in the docs. Which doesn't mean it is not there, I might be looking in the wrong place 😄 

Btw. Great idea this repo!

Cheers,
Dustin Watts

**Comment by cgreening** [comment]
Soon as you use the pins they'll start behaving as normal GPIO pins. So yes - pinMode(39, OUTPUT) should be all you need.

You can also completely disable the JTAG debugging by burning some eFuses.

**Reply by DustinWatts** [reply]
Ow, that is awesomely simple! :D Have to try it out. When I, do I'll report back here!

**Reply by faizannazir** [reply]
Did it work?

**Comment by Basti149** [comment]
I also have another question: could I also use IO39, IO40, IO41 and IO42 as normal outputs/inputs if I deactivate JTag?
Is that right?

**Reply by Volcano-Dragon** [reply]
yes
